src/arrays/arrays_with_methods/hard/q13/MainTest.java

```diff
@@ -0,0 +1,113 @@
+package arrays.arrays_with_methods.hard.q13;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.utils.ArrayUtil;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    static final int n = 10;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = ArrayUtil.arrayToInput(ArrayUtil.generateRandomArray(0, 10000, n));
+        return new Clause[]{
+                new StringLiteral("Enter an array of " + n + " integers:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() {
+        ArrayngeOfPrimes.main(new String[0]);
+    }
+
+    static Stream<int[]> mainMethodInputProvider() {
+        return Stream.of(
+                ArrayUtil.generateAscendingArray(1, n),
+                ArrayUtil.generateRandomArray(1, 10000, n),
+                ArrayUtil.generateRandomArray(1, 10, n)
+        );
+    }
+
+    static Stream<int[]> getPrimesInputProvider() {
+        return Stream.of(
+                ArrayUtil.generateAscendingArray(1, 100),
+                ArrayUtil.generateRandomArray(1, 10000, 35),
+                ArrayUtil.generateRandomArray(1, 10, 10000),
+                ArrayUtil.generateRandomArray(2, 5, 1),
+                ArrayUtil.generateRandomArray(2, 5, 2),
+                ArrayUtil.generateAscendingArray(1, 250, 2),
+                new int[]{5, 7, 5, 2},
+                new int[]{1, 4, 6, 8, 9, 10, 12, 14, 21, 66, 12, 4},
+                new int[]{15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1},
+                new int[]{3169, 1999, 4793, 5783, 7919, 6311, 672832, 733},
+                new int[]{5, 53, 157, 173, 211, 257, 263, 3731367, 1511, 1747, 1753, 1907, 2287, 2417, 3637, 3733, 4013, 4409, 4457, 4597, 4657, 4691, 4993, 5107, 5113, 5393},
```

Comment:
Are these all primes? If so they should not all be, you should throw in some numbers that are one off of being a prime.

Reply:
3731367 is not a prime, I'll throw in one number that is also slightly off